contracts/settling_game/library/library_buildings.cairo

```diff
@@ -0,0 +1,400 @@
+        # Calculate effective building time
+
+        # TODO: Add Max decay to stop overflow
+        let (effective_building_time, _) = unsigned_div_rem(
+            (10000 - decay_rate) * time_balance, 10000
+        )
+
+        return (effective_building_time)
+    end
+
+    func calculate_effective_buildings{
+        syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr
+    }(building_id : felt, time_balance : felt, block_timestamp : felt) -> (
+        effective_buildings : felt
+    ):
+        alloc_locals
+
+        let (base_building_number, time_left) = get_base_building_left(
+            time_balance, block_timestamp
+        )
+
+        let (decay_slope) = get_decay_slope(building_id)
+
+        # pass slope determined by building
+        let (decay_rate) = get_decay_rate(base_building_number, decay_slope)
+
+        # pass actual time balance + decay rate
+        let (effective_building_time) = get_decayed_building_time(time_left, decay_rate)
+
+        let (effective_buildings) = get_effective_buildings(effective_building_time)
+
+        return (effective_buildings)
+    end
+
+    func unpack_buildings{
+        syscall_ptr : felt*,
+        pedersen_ptr : HashBuiltin*,
+        range_check_ptr,
+        bitwise_ptr : BitwiseBuiltin*,
+    }(packed_buildings : PackedBuildings) -> (unpacked_buildings : RealmBuildings):
+        alloc_locals
+
+        let (House) = unpack_data(packed_buildings.housing, 0, 2199023255551)
+        let (StoreHouse) = unpack_data(packed_buildings.economic, 0, 2199023255551)
+        let (Granary) = unpack_data(packed_buildings.economic, 41, 2199023255551)
+        let (Farm) = unpack_data(packed_buildings.economic, 82, 2199023255551)
+        let (FishingVillage) = unpack_data(packed_buildings.economic, 123, 2199023255551)
+        let (Barracks) = unpack_data(packed_buildings.military, 0, 2199023255551)
+        let (MageTower) = unpack_data(packed_buildings.military, 41, 2199023255551)
+        let (ArcherTower) = unpack_data(packed_buildings.military, 82, 2199023255551)
+        let (Castle) = unpack_data(packed_buildings.military, 123, 2199023255551)
+
+        return (
+            unpacked_buildings=RealmBuildings(
+            House=House,
+            StoreHouse=StoreHouse,
+            Granary=Granary,
+            Farm=Farm,
+            FishingVillage=FishingVillage,
+            Barracks=Barracks,
+            MageTower=MageTower,
+            ArcherTower=ArcherTower,
+            Castle=Castle,
+            ),
+        )
+    end
+
+    func pack_buildings{
+        syscall_ptr : felt*,
+        pedersen_ptr : HashBuiltin*,
+        range_check_ptr,
+        bitwise_ptr : BitwiseBuiltin*,
+    }(current_buildings : RealmBuildings, building_id : felt) -> (
+        packed_buildings : PackedBuildings
+    ):
+        alloc_locals
+
+        let (buildings : felt*) = alloc()
+
+        if building_id == RealmBuildingsIds.House:
```

Comment:
The if-else branches in these are the same thing. Is that some relic of a refactor or were they meant to do something different?